### PR TITLE
FIX Run ANALYZE TABLE before fetching table stats

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/.github/issues/290
Replaces https://github.com/silverstripe/silverstripe-framework/pull/11333

Tested on elemental + asset-admin
- https://github.com/creative-commoners/silverstripe-elemental/actions/runs/10363655279
- https://github.com/creative-commoners/silverstripe-asset-admin/actions/runs/10363668130/job/28687862776

### CODE REVIEW

File: src/ORM/Connect/MySQLDatabase.php [hunk 1]
```diff
@@ -566,26 +566,21 @@ public function random()
...
-            $self->query("DELETE FROM \"$table\"");
-            if ($maxID > 0) {
-                $self->query("ALTER TABLE \"$table\" AUTO_INCREMENT = 1");
-            }
-        };
-        if ($this->supportsTransactions()) {
-            $this->withTransaction($fn);
-        } else {
-            $fn();
+        $this->query("DELETE FROM \"$table\"");
```
Comment (GuySartorelli) [inline src/ORM/Connect/MySQLDatabase.php:572]: Need to keep the comment about not using truncate

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Forgot to hit add to review on the above comment, and GitHub doesn't let me not have a comment when I request changes so I have to make a comment even though there's already one and all I want to do is set the status of the PR to reflect the fact that I've requested changes.

File: src/ORM/Connect/MySQLDatabase.php (hunk 1 above)
Comment (emteknetnz) [reply src/ORM/Connect/MySQLDatabase.php:572]: Done

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**